### DISCUSSION TITLE
Unable to close mongoose connection after Jest tests

### DISCUSSION THREAD

**Discussion by Nikola-Milovic** [Automattic/mongoose#11517] [Q&A]
Hello everyone, here's [the link to stack overflow](https://stackoverflow.com/questions/71427088/leaking-mongoose-connection-in-jest-tests) with a lot more details but in general I am creating a mongoose connection inside my App code that I am testing, and then after the tests are finished I try closing the connection but the connection is never closed, always leaking. My doubt is that somehow because these are different contexts, one is the `src` and the other is inside `tests` that maybe the mongoose object is not the same. But when I print the `mongoose.connections.length` I always get 1. And also all of my tests are successful... So not sure what is happening

**Comment by vkarpov15** [comment]
`mongoose.connections.length === 1` is fine, because Mongoose retains the connection object even after the connection is disconnected from the database. Either way, looks like you solved the problem in SO.

**Reply by Nikola-Milovic** [reply]
Oh, that's good to know, thanks
